optuna/integration/botorch.py

```diff
@@ -555,3 +551,31 @@ def sample_independent(
 
     def reseed_rng(self) -> None:
         self._independent_sampler.reseed_rng()
+
+    def after_trial(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        state: TrialState,
+        values: Optional[Sequence[float]],
+    ) -> None:
+        if self._constraints_func is not None:
+            constraints = None
+
+            try:
+                con = self._constraints_func(trial)
+                if not isinstance(con, (tuple, list)):
+                    warnings.warn(
+                        f"Constraints should be a sequence of floats but got {constraints}."
+                    )
+                constraints = tuple(con)
+            except Exception:
+                raise
```

Comment:
If calculated `con` cannot be cast to `tuple`, the error will be raised. How about testing it?

Reply:
Ah, it should be tested already by `test_botorch_constraints_func_invalid_type`.